WcaOnRails/app/views/competitions/_results_table.html.erb

```diff
@@ -37,14 +37,16 @@
   </thead>
 
   <tbody>
+   <% previous_event = "" %>
     <% results.each_with_index do |result, i| %>
       <tr class="sort-by-<%= result.format.sort_by.to_s %> <%= result.muted ? "text-muted" : "" %>">
         <% if !hide_event %>
           <td class="event">
-            <% if !result.muted %>
+            <% if !result.muted && result.eventId != previous_event %>
```

Comment:
Given that we have index already, what about this?
```rb
result.eventId != results[i - 1].eventId
```
This way we don't need to keep track of the previous event on our own =)